app/gui2/src/util/database/reactiveDb.ts

```diff
@@ -68,6 +73,31 @@ export class ReactiveDb<K, V> extends ObservableV2<{
     return this._internal.get(key)
   }
 
+  /**
+   * Checks if the database contains the key.
+   * @param key - A key to chec.
+   */
+  has(key: K): boolean {
+    return this._internal.has(key)
+  }
+
+  /**
+   * Retrieves the value corresponding to a specified key in the database, or insert the provided
+   * one if missing. Will emit an `entryAdded` event after the addition.
+   *
+   * @param key - The key for which to retrieve the value.
+   * @param valueToAdd - Value to add if there's no value under `key`.
+   * @returns The value associated with the key. If was missing, the `valueToAdd` is returned.
+   */
+  getOrAdd(key: K, valueToAdd: V): V {
```

Comment:
I think `valueToAdd` should be a closure `() =>V`, that way you can actually avoid executing some potentially significant code if the key is actually inserted already. That way it can match `setIfUndefined` function from `lib0/map`.